src/leappto/actor_cli.py

```diff
@@ -0,0 +1,154 @@
+import os
+import subprocess
+from argparse import ArgumentParser
+from pprint import pprint
+from subprocess import PIPE
+
+import argcomplete
+import snactor.executors
+
+from snactor import loader
+from snactor import registry
+
+ACTOR_DIRECTORY = '/usr/share/leapp/actors'
+VERSION = "0.2-dev"
+
+
+def _add_identity_options(cli_cmd, context=''):
+    if context:
+        cli_cmd.add_argument('--' + context + '-identity', default=None,
+                             help='Path to private SSH key for the ' + context + ' machine')
+        cli_cmd.add_argument('--' + context + '-user', default="root", help='Connect as this user to the ' + context)
+    else:
+        cli_cmd.add_argument('--identity', default=None, help='Path to private SSH key')
+        cli_cmd.add_argument('--user', '-u', default="root", help='Connect as this user')
+
+
+def _port_spec(arg):
+    """Converts a port forwarding specifier to a (host_port, container_port) tuple
+
+    Specifiers can be either a simple integer, where the host and container port are
+    the same, or else a string in the form "host_port:container_port".
+    """
+    host_port, sep, container_port = arg.partition(":")
+    host_port = int(host_port)
+    if not sep:
+        container_port = host_port
+    else:
+        container_port = int(container_port)
+    return str(host_port), container_port
+
+
+def to_port_map(items):
+    port_map = {}
+    for source, target in items:
+        port_map[source] = port_map.get(source, []) + [target]
+    return {k: list(set(v)) for k, v in port_map.items()}
```

Comment:
key, value looks better imho

Reply:
well k, v is a very common naming scheme in python, especially in a dict comprehension context and it leaves little space for ambiguity.